aten/src/ATen/native/cuda/UnaryOpsKernel.cu

```diff
@@ -65,6 +65,14 @@ void round_kernel_cuda(TensorIterator& iter) {
   });
 }
 
+void trunc_kernel_cuda(TensorIterator& iter) {
+  AT_DISPATCH_FLOATING_TYPES_AND_HALF(iter.dtype(), "trunc_cuda", [&]() {
+    gpu_kernel(iter, []GPU_LAMBDA(scalar_t a) -> scalar_t {
+      return std::trunc(a);
```

Comment:
Are you sure there is no need to overload to call `truncf` ?

Reply:
Yeah, it's overloaded https://en.cppreference.com/w/cpp/numeric/math/trunc

Reply:
If you are worried, here's some evidence. Compile the following program using

nvcc -ptx -src-in-ptx -arch=sm_60 test.cu

```cuda
#include <cuda_runtime.h>

__global__ void test_trunc_f(float x,  float& x2) {
  x2 = std::trunc(x);
}

__global__ void test_trunc_d(double x,  double& x2) {
  x2 = std::trunc(x);
}

__global__ void test_trunc_f_d(float x,  float& x2) {
  x2 = truncf(x);
}
```

The output shows (Note that the first and third functions are compiled to the same asm code)

```
//
// Generated by NVIDIA NVVM Compiler
//
// Compiler Build ID: CL-26907403
// Cuda compilation tools, release 10.1, V10.1.243
// Based on LLVM 3.4svn
//

.version 6.4
.target sm_60
.address_size 64

	// .globl	_Z12test_trunc_ffRf

.visible .entry _Z12test_trunc_ffRf(
	.param .f32 _Z12test_trunc_ffRf_param_0,
	.param .u64 _Z12test_trunc_ffRf_param_1
)
{
	.reg .f32 	%f<3>;
	.reg .b64 	%rd<3>;


	ld.param.f32 	%f1, [_Z12test_trunc_ffRf_param_0];
	ld.param.u64 	%rd1, [_Z12test_trunc_ffRf_param_1];
	cvta.to.global.u64 	%rd2, %rd1;
	cvt.rzi.f32.f32	%f2, %f1;
	st.global.f32 	[%rd2], %f2;
	ret;
}

	// .globl	_Z12test_trunc_ddRd
.visible .entry _Z12test_trunc_ddRd(
	.param .f64 _Z12test_trunc_ddRd_param_0,
	.param .u64 _Z12test_trunc_ddRd_param_1
)
{
	.reg .f64 	%fd<3>;
	.reg .b64 	%rd<3>;


	ld.param.f64 	%fd1, [_Z12test_trunc_ddRd_param_0];
	ld.param.u64 	%rd1, [_Z12test_trunc_ddRd_param_1];
	cvta.to.global.u64 	%rd2, %rd1;
	cvt.rzi.f64.f64	%fd2, %fd1;
	st.global.f64 	[%rd2], %fd2;
	ret;
}

	// .globl	_Z14test_trunc_f_dfRf
.visible .entry _Z14test_trunc_f_dfRf(
	.param .f32 _Z14test_trunc_f_dfRf_param_0,
	.param .u64 _Z14test_trunc_f_dfRf_param_1
)
{
	.reg .f32 	%f<3>;
	.reg .b64 	%rd<3>;


	ld.param.f32 	%f1, [_Z14test_trunc_f_dfRf_param_0];
	ld.param.u64 	%rd1, [_Z14test_trunc_f_dfRf_param_1];
	cvta.to.global.u64 	%rd2, %rd1;
	cvt.rzi.f32.f32	%f2, %f1;
	st.global.f32 	[%rd2], %f2;
	ret;
}
```